src/installer/_core.py

```diff
@@ -0,0 +1,113 @@
+"""Wheel Installation logic, sans I/O
+"""
+
+import posixpath
+
+from installer._compat import StringIO
+from installer._compat.typing import TYPE_CHECKING
+from installer.parsers import parse_metadata_file
+from installer.protocols import Destination, Scheme, WheelSource
+from installer.validation import InvalidWheelSource
+
+if TYPE_CHECKING:
+    from typing import Callable, List, Optional
+
+    from installer._compat.typing import Text
+
+    Validator = Callable[[WheelSource], None]
+
+
+__all__ = ["Installer"]
+
+
+class Installer(object):
+    """Implements wheel installation logic
+
+    Supports Wheel version 1.0 (PEP 427).
+    """
+
+    def __init__(self, name, validators):
+        # type: (Optional[Text], List[Validator]) -> None
+        super(Installer, self).__init__()
+        self.name = name
+        self._validators = validators
+
+    def install(self, source, destination, requested=False):
+        # type: (WheelSource, Destination, bool) -> None
+        """Install a wheel, as described by ``source``, to ``destination``.
+        """
+
+        # Process the WHEEL file
+        wheel_metadata_stream = source.open_dist_info(u"WHEEL")
+        metadata = parse_metadata_file(wheel_metadata_stream)
+
+        # Ensure compatibility with this wheel version.
+        if metadata["Wheel-Version"] != "1.0":
+            raise InvalidWheelSource(
+                source, "Incompatible Wheel-Version: only support version 1.0",
+            )
```

Comment:
```suggestion
        wheel_version = metadata["Wheel-Version"]
        if  wheel_version != "1.0":
            raise InvalidWheelSource(
                source, "Incompatible Wheel-Version: version 1.0 supported, not {}".format(wheel_version),
            )
```
And this makes me miss the walrus operator. 😢